### PR TITLE
support smatrix in caller.jl

### PR DESCRIPTION
**Pull request by tthsqe12**
goes with https://github.com/oscar-system/libsingular-julia/pull/33

### CODE REVIEW

File: src/caller.jl [hunk 1]
```diff
@@ -192,22 +197,23 @@ function prepare_argument(x::sresolution)
 end
 
 function prepare_argument(x::Any)
+    in(:ptr, fieldnames(typeof(x))) || error("unrecognized argument")
```
Comment (fingolfin) [inline src/caller.jl:200]: Not a change request, just wondering: is there any reason not to write this in infix notation, i.e.
```suggestion
    :ptr in fieldnames(typeof(x))) || error("unrecognized argument")
```

File: test/caller-test.jl [hunk 2]
```diff
@@ -89,6 +76,19 @@
...
 end
 
+@testset "caller.LibFinvar" begin
+   Qw, (w,) = FunctionField(QQ, ["w"])
+   K, w = AlgebraicExtensionField(Qw, w^2+w+1)
+   R, (x, y, z) = PolynomialRing(K, ["x","y","z"])
+   M1 = Matrix(R, [0 1 0; 0 0 1; 1 0 0])
+   M2 = Matrix(R, [1 0 0; 0 w 0; 0 0 w^2])
+   P, S, IS = Singular.LibFinvar.invariant_ring(M1, M2)
+   @assert isa(P, Singular.smatrix{Singular.spoly{Singular.n_algExt}})
```
Comment (fingolfin) [inline test/caller-test.jl:86]: Same as above: how about this?
```suggestion
   @assert P ia Singular.smatrix{Singular.spoly{Singular.n_algExt}}
```

**Review by fingolfin — APPROVED** [review]
Looks good to me, minor question

File: src/caller.jl (hunk 1 above)
Comment (tthsqe12) [reply src/caller.jl:200]: There is no reason except that I do not know/do not want to know the precedences: is `a in b || c` the same as `(a in b) || c` or `a in (b || c)`? :)

Comment (thofma) [reply src/caller.jl:200]: Even with parenthesis it is one comma less!

File: test/caller-test.jl [hunk 3]
```diff
@@ -89,6 +76,19 @@
...
+   @assert typeof(P) == typeof(S) == typeof(IS)
+   @assert nrows(P) == 1
+   @assert ncols(P) == 3
```
Comment (thofma) [inline test/caller-test.jl:89]: I think these should all be `@test`.

Comment (tthsqe12) [reply test/caller-test.jl:89]: oops

File: src/caller.jl (hunk 1 above)
Comment (tthsqe12) [reply src/caller.jl:200]: so I just checked it out. Just barely works!
```
julia> Base.operator_precedence(:+)
11
julia> Base.operator_precedence(:in)
7
julia> Base.operator_precedence(:*)
12
julia> Base.operator_precedence(:||)
5
```

Comment (fingolfin) [reply src/caller.jl]: It is not about the number of characters, but rather readability: `a in b` to me is easier to grok than `in(a,b)`, same for`isa`

**Pull request merged**